webdriver/tests/bidi/script/add_preload_script/add_preload_script.py

```diff
@@ -170,3 +170,75 @@ async def test_page_script_can_access_preload_script_properties(
         await_promise=True,
     )
     assert result == {"type": "number", "value": 42}
+
+
+@pytest.mark.asyncio
+async def test_add_preload_script_in_iframe_with_top_context_specified(
+    bidi_session, add_preload_script, new_tab, test_page_same_origin_frame
+):
+    await add_preload_script(function_declaration="() => { window.bar='foo'; }", contexts=[new_tab["context"]])
+
+    await bidi_session.browsing_context.navigate(
+        context=new_tab["context"],
+        url=test_page_same_origin_frame,
+        wait="complete",
+    )
+
+    # Check that preload script applied the changes to the window
+    result = await bidi_session.script.evaluate(
+        expression="window.bar",
+        target=ContextTarget(new_tab["context"]),
+        await_promise=True,
+    )
+    assert result == {"type": "string", "value": "foo"}
+
+    contexts = await bidi_session.browsing_context.get_tree(root=new_tab["context"])
+
+    assert len(contexts[0]["children"]) == 1
+    frame_context = contexts[0]["children"][0]
+
+    # Check that preload script applied the changes to the iframe
+    result = await bidi_session.script.evaluate(
+        expression="window.bar",
+        target=ContextTarget(frame_context["context"]),
+        await_promise=True,
+    )
+    assert result == {"type": "string", "value": "foo"}
+
+
+@pytest.mark.asyncio
+async def test_page_script_context_isolation(
+    bidi_session, add_preload_script, top_context, new_tab, test_page
+):
+    await add_preload_script(
+        function_declaration="() => { window.baz = 42; }",
+        contexts=[top_context['context']]
```

Comment:
What should actually happen when we specify the same context multiple times? Should we try to fail on that as well, or ignore? I think that we missed that in the BiDi spec.

Reply:
Good question. There is no specific part in the spec if this situation. My interpenetration of the spec atm would be that any duplicates would just be ignored as the only place we check is when we try to run the `preload script` and check if the Document's top level browsing context is included in the list.

Reply:
We could assume that for now, yes. Would you mind adding a test for that as well/ Maybe as tentative and reference a BiDi issue in a comment?

Reply:
I will open a separate PR, that way we can merge this.

Reply:
It would be still good to get it included in this PR given that updates are needed anyway. And it's a simple test with a preload script that eg. increments a value attached on the window.